storage/replica_raftstorage.go

```diff
@@ -193,8 +192,13 @@ func (r *Replica) loadAppliedIndex(eng engine.Engine) (uint64, error) {
 		return 0, err
 	}
 	if v != nil {
-		var err error
-		_, appliedIndex, err = encoding.DecodeUint64(v.GetRawBytes())
+		bytes, err := v.GetBytes()
+		if err != nil {
+			return 0, err
+		}
+		// Cannot use v.GetInt() here because we need the precision of
+		// uint64; GetInt() is lossy.
```

Comment:
How is `GetInt()` lossy? It decodes using `encoding.DecodeUint64()` and casts to an `int64`. I think you should be able to cast it back to a `uint64` if desired.